module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -467,14 +467,14 @@ func testAddingEntities(t *testing.T, pool *Pool, entitiesToBeAdded []*unittest.
 		if ejectionMode == RandomEjection {
 			if i >= len(pool.poolEntities) {
 				require.True(t, slotAvailable)
-				require.True(t, ejectionHappened)
+				require.NotNil(t, ejectedEntity)
```

Comment:
Similarly, to ensure the reliability of the process, in `RandomEjection` mode, we should check that (1) the `ejectedEntity` indeed belongs to the `entitiesToBeAdded`, and (2) it is a unique entity.

Reply:
https://github.com/onflow/flow-go/pull/3961/commits/9f4d76d8ab4fb02b54a5ecccc58c303944ffb50f